tests/test_main.py

```diff
@@ -234,3 +234,17 @@ def test_generate_column_headers(tmp_path):
         assert columns == next(csv_reader)
         # we want only a header, no row data
         assert not next(csv_reader, None)
+
+
+def test_generate_column_headers_fails_with_star(tmp_path):
+    # Ensure that a helpful error message is generated
+    # if a user tries to generate columns from a query
+    # that has `*` in the select list
```

Comment:
Doesn't the query embedded in `test_generate_column_headers` have `*` in the select list?

Reply:
It does, and my initial query was possible a bit too clever for its own good, e.g.

```sql
;WITH cte as (
  SELECT
    foo,
    bar
  FROM baz)
SELECT *
FROM cte
```

the "main" select outside of the CTE uses `*` in the column list  but it's possible to extract the explicitly defined column list from the CTE's definition. This is what `qualify_columns()` does (the same is true for subqueries etc.). However in the case of

```sql
SELECT * FROM Table
```

Without a schema definition (which it is possible to extract from `tpp_schema.py` in `ehrql` (although that's almost certainly veering into lily-gilding territory), `sqlglot` doesn't know what columns `*` pertains to.